frontend/src/renderer/components/Process/Tabs/SelectDest.svelte

```diff
@@ -1,19 +1,36 @@
 <script lang="ts">
   import FileSelector from "@components/FileSelector.svelte";
-  import { FolderPlusIcon, FolderIcon } from "svelte-feather-icons";
+  import TextInputRow from "@components/TextInputRow.svelte";
+  import { FolderPlusIcon, FolderIcon, FileTextIcon } from "svelte-feather-icons";
   import { currentPage, processState } from "@util/stores";
+  import Card from "@components/Card.svelte";
+  
   let filePaths = [];
   $: console.log(filePaths);
 
   $: if (filePaths) {
-    $processState.destDir = filePaths[0];
+    processState.update(state => ({
+      ...state,
+      destDir: filePaths[0]
+    }));
   }
+
+  $: if ($processState.destFileName) {
+    const sanitized = $processState.destFileName.replace(/[/\\?%*:|"<>]/g, '-');
+    if (sanitized !== $processState.destFileName) {
+       processState.update(state => ({ ...state, destFileName: sanitized }));
+    }
+  }
+
   $: if (
     $processState.destDir &&
     $processState.destDir?.length > 1 &&
     !$processState.completedTabs[2]
   ) {
-    $processState.completedTabs[2] = true;
+    processState.update(state => ({
+      ...state,
+      completedTabs: state.completedTabs.map((completed, i) => i === 2 ? true : completed)
+    }));
```

Comment:
I think there should be a helper function for setting a tab completed. Having this map code copied across the codebase is a little jarring.